overlord/snapstate/snapstate_update_test.go

```diff
@@ -11896,3 +11896,84 @@ func (s *snapmgrTestSuite) TestUpdateManyFilteredNotAutoRefreshNoRetry(c *C) {
 	c.Check(storeSnapIDs["some-snap-id"], Equals, true)
 	c.Check(storeSnapIDs["some-other-snap-id"], Equals, true)
 }
+
+func (s *snapmgrTestSuite) TestUpdateSnapdAndSnapPullingNewBase(c *C) {
```

Comment:
This is a very nice solution to test the scenario! I had a thorough look and cannot see any issues.